packages/manager/src/features/CloudPulse/shared/CloudPulseTagsFilter.tsx

```diff
@@ -0,0 +1,133 @@
+import { Autocomplete } from '@linode/ui';
+import React from 'react';
+
+import { useAllLinodesQuery } from 'src/queries/linodes/linodes';
+import { themes } from 'src/utilities/theme';
+
+import { TAGS } from '../Utils/constants';
+import { FILTER_CONFIG } from '../Utils/FilterConfig';
+
+import type { FilterValue, Linode } from '@linode/api-v4';
+
+export interface CloudPulseTags {
+  label: string;
+}
+
+export interface CloudPulseTagsSelectProps {
+  defaultValue?: Partial<FilterValue>;
+  disabled?: boolean;
+  handleTagsChange: (tags: CloudPulseTags[], savePref?: boolean) => void;
+  label: string;
+  placeholder?: string;
+  resourceType: string | undefined;
+  savePreferences?: boolean;
+}
+
+export const CloudPulseTagsSelect = React.memo(
+  (props: CloudPulseTagsSelectProps) => {
+    const {
+      defaultValue,
+      handleTagsChange,
+      label,
+      placeholder,
+      resourceType,
+      savePreferences,
+    } = props;
+
+    const { data: linodes, isError, isLoading } = useAllLinodesQuery();
+    // fetch all linode instances, consume the associated tags
+    const tags = React.useMemo(() => {
+      if (!linodes) {
+        return [];
+      }
+      return Array.from(
+        new Set(linodes.flatMap((linode: Linode) => linode.tags))
+      )
+        .sort()
+        .map((tag) => ({ label: tag })) as CloudPulseTags[];
+    }, [linodes]);
+
+    const [selectedTags, setSelectedTags] = React.useState<CloudPulseTags[]>();
+
+    const isAutocompleteOpen = React.useRef(false); // Ref to track the open state of Autocomplete
+
+    const isOptionalFilter = resourceType
```

Comment:
@ankita-akamai , Can't this be passed as a property, we will iterating the same while constructing the props for this right